### PR TITLE
OCPEDGE-2259: feat(jira): add template system with inheritance and integrate with commands

### PR DESCRIPTION
**Pull request by jaypoulz**
- Consolidated 6 type-specific skills into unified template-driven create-issue skill
- Created 13 templates with field-level placeholder merging (child overrides parent fields)
- Consolidated additional_context field into base template (-59 duplicate lines)
- Standardized h4/h5 heading structure across all templates
- Created 6 educational guides (docs/issue-types/) with centralized template reference
- Added /jira:template command with validation script (all 12 templates pass)

MCP Setup: Phase 0 automatic availability detection, guided setup via mcp.json,
tracking in jira-config.json, reconnection troubleshooting, setup guide (MCP_SETUP.md)

Template validation: validate_template.py checks YAML syntax, schema, placeholder
references, inheritance cycles, and redundant overrides

## What this PR does / why we need it:
This PR overhauls Jira creation using Claude by providing customizable, opinionated templates for each issue type.

## Summary by CodeRabbit

* **New Features**
  * Jira CLI: /jira:create supports template, overrides, priority and security options; added /jira:template and a unified template-driven create workflow.

* **Documentation**
  * New template docs, schema, setup and issue-type guides; legacy per-type skill guides consolidated into template-management guidance.

* **Chores**
  * Added many common, project and user templates plus project override configs and a template validation utility.

* **Tests**
  * Validator tests covering syntax, inheritance, security and edge cases.

### CODE REVIEW

File: PLUGINS.md [hunk 1]
```diff
@@ -155,12 +155,14 @@ A plugin to automate tasks with Jira
...
-- **`/jira:create` `<type> [project-key] <summary> [--component <name>] [--version <version>] [--parent <key>]`** - Create Jira issues (story, epic, feature, task, bug, feature-request) with proper formatting
+- **`/jira:create` `<type> [project-key] <summary> [--component <name>] [--version <version>] [--parent <key>] [--template <name>]`** - Create Jira issues (story, epic, feature, task, bug, feature-request) with proper formatting
 - **`/jira:generate-feature-doc` `<feature-key>`** - Generate comprehensive feature documentation from Jira feature and all related issues and PRs
 - **`/jira:generate-test-plan` `[JIRA issue key] [GitHub PR URLs]`** - Generate test steps for a JIRA issue
 - **`/jira:grooming` `[project-filter] [time-period] [--component component-name] [--label label-name] [--type issue-type] [--status status] [--story-points]`** - Analyze new bugs and cards added over a time period and generate grooming meeting agenda
+- **`/jira:search` `<jql-query> [--display-format template|full|custom:fields] [--display-template template-name] [--created-with-template template-name]`** - Search JIRA issues with customizable display formats
```
Comment (jaypoulz) [inline PLUGINS.md]: Remove search, this was from an artifact of making sure display format was integrated.

File: docs/data.json [hunk 2]
```diff
@@ -114,6 +114,12 @@
           "name": "grooming",
           "synopsis": "/jira:grooming [project-filter] [time-period] [--component component-name] [--label label-name] [--type issue-type] [--status status] [--story-points]"
         },
+        {
+          "argument_hint": "<jql-query> [--display-format template|full|custom:fields] [--display-template template-name] [--created-with-template template-name]",
```
Comment (jaypoulz) [inline docs/data.json]: search as a command doesn't exist as part of this pr, so remove this.

File: docs/data.json [hunk 3]
```diff
@@ -193,6 +205,11 @@
           "id": "jira-doc-generator",
           "name": "Jira Feature Documentation Generator"
         },
+        {
+          "description": "Implementation guide for searching JIRA issues with customizable template-based display formats",
```
Comment (jaypoulz) [inline docs/data.json]: remove this.

File: plugins/jira/README.md [hunk 4]
```diff
@@ -16,45 +18,122 @@ Comprehensive Jira integration for Claude Code, providing AI-powered tools to an
...
+## Common Conventions
 
-For your Jira token, use https://issues.redhat.com/secure/ViewProfile.jspa?selectedTab=com.atlassian.pats.pats-plugin:jira-user-personal-access-tokens
+### Jira Formatting
 
-### Notes and tips
+**Heading Standards:**
 
-- Do not commit real tokens. If you must keep a project-local file, prefer committing a `mcp.json.sample` with placeholders, and keep your real `mcp.json` untracked.
-- Consider using the [rh-pre-commit](https://source.redhat.com/departments/it/it_information_security/leaktk/leaktk_components/rh_pre_commit) hook to scan for secrets accidentally left in commits.
```
Comment (jaypoulz) [inline plugins/jira/README.md:45]: These two lines should probably have a home in the new README.

File: plugins/jira/README.md [hunk 5]
```diff
@@ -16,45 +18,122 @@ Comprehensive Jira integration for Claude Code, providing AI-powered tools to an
...
-- The `atlassian` server example uses an MCP container image: `ghcr.io/sooperset/mcp-atlassian:latest`.
-- If you prefer Docker, replace the `podman` command with `docker` (arguments are typically the same).
-- If Podman is installed via Podman Machine on macOS, ensure it is running: `podman machine start`.
-- Keep `JIRA_SSL_VERIFY` as "true" unless you have a specific reason to disable TLS verification.
```
Comment (jaypoulz) [inline plugins/jira/README.md:49]: Add this to the generated json.

File: plugins/jira/README.md [hunk 6]
```diff
@@ -240,6 +320,41 @@ See [commands/create-release-note.md](commands/create-release-note.md) for full
...
+**Common templates** (work with any project):
+- `common-story` - User stories with acceptance criteria
+- `common-epic` - Epics with scope and timeline
+- `common-bug` - Bug reports with reproduction steps
+- `common-spike` - Research and investigation
+- `common-task` - Technical work and operational tasks
+- `common-feature` - Strategic features with market analysis
+
+**Product-specific templates:**
+- Templates for OCPBUGS, RHEL, and other product organizations
```
Comment (jaypoulz) [inline plugins/jira/README.md:421]: `product-level organizations`

**Review by jaypoulz — COMMENTED** [review]
self-review in progress

File: plugins/jira/templates/README.md [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+Templates are organized by issue type. Each issue type has an educational guide with best practices.
+
+| Issue Type | Guide | Available Templates |
+|------------|-------|---------------------|
+| Story | [Story Guide](../docs/issue-types/story.md) | `common-story` (all projects) |
+| Epic | [Epic Guide](../docs/issue-types/epic.md) | `common-epic` (all projects) |
+| Task | [Task Guide](../docs/issue-types/task.md) | `common-task` (all projects) |
+| Spike | [Spike Guide](../docs/issue-types/spike.md) | `common-spike`, `ocpedge-spike` |
+| Bug | [Bug Guide](../docs/issue-types/bug.md) | `common-bug`, `ocpbugs-bug`, `osdocs-bug`, `rhel-bug` |
+| Feature | [Feature Guide](../docs/issue-types/feature.md) | `common-feature`, `extended-feature`, `rfe` |
```
Comment (jaypoulz) [inline plugins/jira/templates/README.md]: seperate guide for RFE

File: plugins/jira/templates/README.md [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+└── common/rfe.yaml                 # RFE feature requests
+```
+
+## Template Reference Guide
+
+Templates are organized by issue type. Each issue type has an educational guide with best practices.
+
+| Issue Type | Guide | Available Templates |
+|------------|-------|---------------------|
+| Story | [Story Guide](../docs/issue-types/story.md) | `common-story` (all projects) |
```
Comment (jaypoulz) [inline plugins/jira/templates/README.md:110]: reorder commits since these links won't work

File: plugins/jira/templates/README.md [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+# Jira Issue Templates
+
+Pre-built templates for creating consistent Jira issues across projects and teams.
+
+## Template Inheritance
+
+Templates use inheritance to reduce duplication:
+
+```
+_base.yaml                           # Shared defaults
```
Comment (jaypoulz) [inline plugins/jira/templates/README.md]: This table doesn't really help explain how templates inherit values

File: plugins/jira/templates/README.md [hunk 10]
```diff
@@ -0,0 +1,170 @@
...
+/jira:create story OCPEDGE "Add authentication"
+
+# List available templates
+/jira:template list
+
+# Create with specific template
+/jira:create bug OCPBUGS "API server crash" --template ocpbugs-bug
+```
+
+## Directory Structure
```
Comment (jaypoulz) [inline plugins/jira/templates/README.md]: This directory structure can be simplified

File: plugins/jira/templates/README.md [hunk 11]
```diff
@@ -0,0 +1,170 @@
...
+   ```
+
+2. **Test and refine:** Use the template, iterate
+
+3. **Publish (optional):**
+   ```bash
+   /jira:template publish my-story
+   ```
+
+## Project-Specific Fields
```
Comment (jaypoulz) [inline plugins/jira/templates/README.md]: This whole section can be removed entirely

File: plugins/jira/templates/SCHEMA.md [hunk 12]
```diff
@@ -0,0 +1,522 @@
...
+
+## Overview
+
+Templates drive the entire issue creation workflow:
+- Define required fields and their prompts
+- Generate interactive collection workflows
+- Provide validation rules
+- Specify formatting and defaults
+
+## Template File Structure
```
Comment (jaypoulz) [inline plugins/jira/templates/SCHEMA.md:15]: We should annotate what some of these lines mean from content perspective

File: plugins/jira/templates/SCHEMA.md [hunk 13]
```diff
@@ -0,0 +1,522 @@
...
+
+# Validation rules (optional)
+validation:
+  summary_max_length: 100
+  summary_should_start_with_verb: true
+  required_fields:
+    - field_name
+```
+
+## Placeholder Schema
```
Comment (jaypoulz) [inline plugins/jira/templates/SCHEMA.md:227]: This probably needs an example to explain this better

File: plugins/jira/templates/SCHEMA.md [hunk 14]
```diff
@@ -0,0 +1,522 @@
...
+
+Each placeholder in the `placeholders` array defines one piece of information to collect from the user.
+
+### Core Fields
+
+#### `name` (required)
```
Comment (jaypoulz) [inline plugins/jira/templates/SCHEMA.md:248]: These schemas are difficult to understand. Consider reworking

File: plugins/jira/templates/SCHEMA.md [hunk 15]
```diff
@@ -0,0 +1,522 @@
...
+- **Description:** For placeholders that expand to template content
+- **Used with:** `placeholder_content`
+- **Example:**
+  ```yaml
+  - name: additional_spikes_link
+    placeholder_name: additional_spikes_placeholder
+    placeholder_content: "* [Additional Spikes|https://...]"
+  ```
+
+## Template Inheritance
```
Comment (jaypoulz) [inline plugins/jira/templates/SCHEMA.md]: This probably needs to be the first block in the templates guide

File: plugins/jira/templates/SCHEMA.md [hunk 16]
```diff
@@ -0,0 +1,522 @@
...
+```
+
+### Inverted Sections
+```yaml
+{{^section_name}}
+  Content shown if section_name is falsy
+{{/section_name}}
+```
+
+## Complete Example
```
Comment (jaypoulz) [inline plugins/jira/templates/SCHEMA.md:833]: This complete example should be simplified

File: plugins/jira/templates/SCHEMA.md [hunk 17]
```diff
@@ -0,0 +1,522 @@
...
+
+3. **On errors**: Point to relevant guidance
+   ```
+   Epic seems too large (>5 sprints). See docs/issue-types/epic.md#anti_patterns
+   for guidance on epic sizing.
+   ```
+
+## Reference Docs
+
+For educational content about each issue type:
```
Comment (jaypoulz) [inline plugins/jira/templates/SCHEMA.md:886]: This already exists as a table somewhere

File: plugins/jira/templates/common/bug.yaml [hunk 18]
```diff
@@ -0,0 +1,180 @@
...
+      - Contrast with actual results
+    examples:
+      - "The kube-apiserver pod should start successfully after the upgrade. The pod should be in Running state with 1/1 containers ready. API requests to the cluster should succeed normally."
+      - "The button should respond to tap with visual feedback and navigate to the credentials input screen."
+    validation:
+      min_length: 10
+      max_length: 2000
+
+validation:
+  summary_max_length: 120
```
Comment (jaypoulz) [inline plugins/jira/templates/common/bug.yaml]: I don't understand how this ended up being seperate

File: plugins/jira/templates/common/epic.yaml [hunk 19]
```diff
@@ -0,0 +1,166 @@
...
+        description: "Extra Large - About 5 sprints"
+    default: "M"
+    help_text: |
+      Size helps with capacity planning and sprint allocation.
+      Based on estimated number of sprints to complete.
+      If epic is >5 sprints, consider splitting or creating as Feature.
+      If epic is <1 sprint, consider creating as Story instead.
+
+validation:
+  summary_max_length: 80
```
Comment (jaypoulz) [inline plugins/jira/templates/common/epic.yaml]: Summary max length should be set in the base template. I don't see why these is set for different templates.

File: plugins/jira/templates/common/ocpbugs-bug.yaml [hunk 20]
```diff
@@ -0,0 +1,121 @@
...
+    description: "Platform (AWS/Azure/GCP/vSphere/BareMetal/etc.)"
+    required: false
+    default: "Not specified"
+
+  - name: cluster_type
+    description: "Cluster type (SNO/Compact/Standard/HyperShift/etc.)"
+    required: false
+    default: "Standard"
+
+jira_fields:
```
Comment (jaypoulz) [inline plugins/jira/templates/common/ocpbugs-bug.yaml:162]: what are jira_fields in a template?

File: plugins/jira/templates/common/story.yaml [hunk 21]
```diff
@@ -0,0 +1,133 @@
...
+    examples:
+      - "Test that node pools scale up when CPU exceeds 80%"
+      - "Demonstrate that scaling policies can be configured via CLI"
+      - "Verify that when a cluster admin sets max nodes to 10, the pool never exceeds 10 nodes"
+      - "Given CPU utilization is at 85%, when the policy is active, then a new node is provisioned within 2 minutes"
+
+# Story-specific validation
+validation:
+  summary_max_length: 100
+  summary_should_not_contain:
```
Comment (jaypoulz) [inline plugins/jira/templates/common/story.yaml]: Seems ultra-specific.

**Review by jaypoulz — COMMENTED** [review]
Review for template commit

File: plugins/jira/skills/cntrlplane/SKILL.md [hunk 22]
```diff
@@ -28,7 +28,7 @@ This skill is automatically invoked by the `/jira:create` command when the proje
```
Comment (jaypoulz) [inline plugins/jira/skills/cntrlplane/SKILL.md]: why does this skill still exist?

File: plugins/jira/skills/ocpbugs/SKILL.md [hunk 23]
```diff
@@ -28,7 +28,7 @@ This skill is automatically invoked by the `/jira:create` command when the proje
```
Comment (jaypoulz) [inline plugins/jira/skills/ocpbugs/SKILL.md]: This skill should not exist.

File: plugins/jira/skills/template-management/SKILL.md [hunk 24]
```diff
@@ -0,0 +1,257 @@
...
+✓ YAML syntax valid
+✓ Schema validation passed
+  - name: ocpedge-spike
+  - issue_type: Spike
+  - 5 placeholders defined
+✓ Placeholder references valid
+  Uses: overview, research_questions, involved_teams, ...
+  All defined
+✓ Inheritance chain valid
+  spike.yaml → spike.yaml → _base.yaml
```
Comment (jaypoulz) [inline plugins/jira/skills/template-management/SKILL.md]: confusing, use folder prefixes for this chain

File: plugins/jira/docs/issue-types/epic.md [hunk 25]
```diff
@@ -0,0 +1,498 @@
...
+- Identify who benefits
+- Be concise and direct
+
+**Writing Guidelines:**
+- **Be concise**: Remove unnecessary adjectives and adverbs
+- **Be direct**: State facts, not opinions or emphasis
+- **Be specific**: Use concrete nouns and verbs
+- **Avoid filler words**: "really", "very", "easily", "simply", "just", "actually", "comprehensive", "powerful"
+
+**Good example (concise, direct):**
```
Comment (jaypoulz) [inline plugins/jira/docs/issue-types/epic.md:97]: I'm not a big fan of good and bad. Let's update this and add a policy for using strong and weak instead. This removes the implications of an moral assessments.

File: plugins/jira/README.md (hunk 4 above)
Comment (jaypoulz) [inline plugins/jira/README.md:45]: Let's add instructions for integrating this hook and updating a config value one it's set up.

File: plugins/jira/docs/MCP_SETUP.md [hunk 26]
```diff
@@ -0,0 +1,516 @@
...
+      "env": {
+        "JIRA_URL": "https://issues.redhat.com",
+        "JIRA_USERNAME": "your-email@example.com"
+      }
+    }
+  }
+}
+```
+
+### Multiple Jira Instances
```
Comment (jaypoulz) [inline plugins/jira/docs/MCP_SETUP.md]: Omit this entirely.

File: plugins/jira/docs/MCP_SETUP.md [hunk 27]
```diff
@@ -0,0 +1,516 @@
...
+
+## Advanced Configuration
+
+### Custom Container Image
+
+If you've built a custom atlassian MCP server image:
+
+```json
+{
+  "mcpServers": {
```
Comment (jaypoulz) [inline plugins/jira/docs/MCP_SETUP.md]: this file spams this structure. Simplify these blocks

File: plugins/jira/docs/issue-types/bug.md [hunk 28]
```diff
@@ -0,0 +1,659 @@
...
+Guide on Jira Bugs: how to write effective bug reports with complete information for resolution.
+
+## Using Templates
+
+See [Template Reference Guide](../../templates/README.md#template-reference-guide) for complete template→guide mapping.
+
+**Create a bug:**
+```bash
+# Auto-selects appropriate template
+/jira:create bug OCPBUGS "API server crashes on node deletion"
```
Comment (bryan-cox) [inline plugins/jira/docs/issue-types/bug.md:11]: I wonder if we can just drop the OCPBUGS here 🤔 

If a user runs this `/jira:create bug ...` wouldn't the project always be OCPBUGS.

File: plugins/jira/docs/issue-types/epic.md [hunk 29]
```diff
@@ -0,0 +1,498 @@
...
+Guide on Jira Epics: what they are, how to use them, and best practices.
+
+## Using Templates
+
+See [Template Reference Guide](../../templates/README.md#template-reference-guide) for complete template→guide mapping.
+
+**Create an epic:**
+```bash
+# Auto-selects appropriate template
+/jira:create epic OCPEDGE "Multi-cluster metrics aggregation for ROSA HCP"
```
Comment (bryan-cox) [inline plugins/jira/docs/issue-types/epic.md]: Should this obfuscate the project or update the comment so it knows this is just an example vs all epics should be made under OCPEDGE?

**Review by bryan-cox — COMMENTED** [review]
Took a first pass

File: plugins/jira/docs/issue-types/epic.md (hunk 29 above)
Comment (bryan-cox) [reply plugins/jira/docs/issue-types/epic.md]: This comment applies to several other areas as well.

File: plugins/jira/docs/issue-types/bug.md [hunk 30]
```diff
@@ -0,0 +1,659 @@
...
+
+### 7. Environment Details
+
+Version, platform, configuration where bug occurs
+
+### 8. Additional Context
+
+Logs, screenshots, error messages, workarounds
+
+## Bug Report Template
```
Comment (jaypoulz) [inline plugins/jira/docs/issue-types/bug.md]: The bug report template is captured in the template. Let's remove this.

File: plugins/jira/docs/issue-types/bug.md (hunk 28 above)
Comment (jaypoulz) [reply plugins/jira/docs/issue-types/bug.md:11]: I wrestled with this. I think the answer is usually, but not always.
Product level bugs could also belong to OSDOCS (openshift docs) or RHEL
It's also sometimes the case for users to want to file a bug on their personal project. We have bugs that aren't openshift bugs for things like our two-node-toolbox repo.

File: plugins/jira/docs/issue-types/epic.md (hunk 25 above)
Comment (jaypoulz) [reply plugins/jira/docs/issue-types/epic.md:97]: I thought about this more and I couldn't come up with anything that doesn't have some kind of moral assessement. Good and bad are as good as anything else I thought of, and simpler.

File: plugins/jira/docs/issue-types/bug.md (hunk 28 above)
Comment (jaypoulz) [reply plugins/jira/docs/issue-types/bug.md:11]: We could add a default though if you omit it.

File: plugins/jira/docs/issue-types/epic.md (hunk 29 above)
Comment (jaypoulz) [reply plugins/jira/docs/issue-types/epic.md]: Good catch. :) I'll run a scan for team-specific examples.
